src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java

```diff
@@ -0,0 +1,156 @@
+package org.opentestsystem.ap.imrt.common.model;
+
+import org.hibernate.annotations.Type;
+import org.hibernate.validator.constraints.NotBlank;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.time.Instant;
+import java.util.List;
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@Table(name = "item")
+@EntityListeners(AuditingEntityListener.class)
+@DiscriminatorColumn(name = "classification")
+public abstract class BaseItem extends BaseEntity {
+    @Id
+    @Column
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer key;
+
+    @NotBlank
+    private String id;
+
+    private String subject = "";
+
+    private String grade = "";
+
+    @NotBlank
+    private String workflowStatus;
+
+    @NotBlank
+    @Column(name = "item_type")
+    private String itemType;
+
+    @Column(name = "dok")
+    private String depthOfKnowledge = "";
+
+    @NotBlank
+    private String itemCreatedBy;
+
+    @Basic
+    private Instant itemCreatedAt;
+
+    @Type(type = "jsonb")
+    @Column(columnDefinition = "jsonb")
+    @NotNull
+    private org.opentestsystem.ap.common.model.Item itemJson;
+
+    @OneToMany
+    @JoinColumn(name = "item_key")
+    @OrderBy("type ASC")
+    private List<StandardId> standardIds;
+
+    public List<StandardId> getStandardIds() {
+        return standardIds;
+    }
+
+    public void setStandardIds(final List<StandardId> standardIds) {
+        this.standardIds = standardIds;
+    }
+
+    public Integer getKey() {
+        return key;
+    }
+
+    public void setKey(final Integer key) {
+        this.key = key;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(final String id) {
+        this.id = id;
+    }
+
+    public String getSubject() {
+        return subject;
+    }
+
+    public void setSubject(final String subject) {
+        this.subject = subject;
+    }
+
+    public String getGrade() {
+        return grade;
+    }
+
+    public void setGrade(final String grade) {
```

Comment:
nit: do you really want your setters to be public and your models to be mutable?

Reply:
How did you manage updates to items within JPA without the ability to modify the object directly?  It seems like JPA (hibernate) doesn't like immutablity by default

Reply:
I'm curious about that myself.  We had that problem in TDS - we added public setters so that JPA could hydrate the objects.  We put all the setters at the bottom of the code file (which goes against normal convention, I know) with a big comment that said "DO NOT USE THESE - they're only for hibernate" (paraphrasing).  If there's another way to do it & maintain immutability (like tell JPA/Hibernate to use a builder?) I'd love to know what it is.

Reply:
I don't think Will is suggesting true immutability. Just "hide" the setters so only JPA can get at them.

Reply:
We used builders with a copy() method to create a new instance with updated values.

Reply:
Understood, and yes Will.